lib/Macros/Sources/SwiftMacros/DebugDescriptionMacro.swift

```diff
@@ -0,0 +1,287 @@
+        var computedProperties: Set<String> = []
+        for property in properties.values {
+            if isComputedProperty(property), let name = bindingName(property) {
+                computedProperties.insert(name)
+            }
+        }
+
+        // Iterate the string's segments, and convert property expressions into LLDB variable references.
+        var summarySegments: [String] = []
+        for segment in descriptionString.segments {
+            switch segment {
+            case let .stringSegment(segment):
+                summarySegments.append(segment.content.text)
+            case let .expressionSegment(segment):
+                guard let labeledExpr = segment.expressions.single, labeledExpr.label == nil else {
+                    // This catches `appendInterpolation` overrides.
+                    let message: ErrorMessage = "unsupported custom string interpolation expression"
+                    context.diagnose(node: segment, message: message)
+                    return []
+                }
+
+                let expr = labeledExpr.expression
+
+                var propertyChain: [String] = []
+                if let declRef = expr.as(DeclReferenceExprSyntax.self) {
+                    // A reference to a single property on self.
+                    propertyChain.append(declRef.baseName.text)
+                } else if let memberAccess = expr.as(MemberAccessExprSyntax.self) {
+                    // A chain of properties, possibly starting with an explicit self.
+                    enumerateMembers(memberAccess) { declRef in
+                        propertyChain.append(declRef.baseName.text)
+                    }
+                } else {
+                    // The expression was neither a DeclReference nor a MemberAccess.
+                    let message: ErrorMessage = "unsupported expression; stored properties only"
+                    context.diagnose(node: expr, message: message)
+                    return []
+                }
+
+                // Explicit self are removed before use by LLDB.
+                if propertyChain[0] == "self" {
+                    propertyChain.removeFirst()
+                }
+
+                // Check that the root property is not a computed property on `self`.
+                // Ideally, all properties would be checked, but a macro expansion can
+                // at best only check the properties of the type it's attached to.
+                let rootProperty = propertyChain[0]
+                guard !computedProperties.contains(rootProperty) else {
+                    let message: ErrorMessage = "cannot reference computed properties"
+                    context.diagnose(node: expr, message: message)
+                    return []
+                }
+
+                let propertyPath = propertyChain.joined(separator: ".")
+                summarySegments.append("${var.\(propertyPath)}")
+            @unknown default:
+                return []
```

Comment:
Should you diagnose this?

Reply:
Maybe an "unexpected internal error" would be enough, so we don't deceive the user to think the macro succeeded.